src/llm/generate_task_plan/prompt.py

```diff
@@ -5,23 +5,54 @@
     SystemMessagePromptTemplate,
 )
 
+SUBQUESTIONS_TEMPLATE = """
+You are {name}, an {role}.
+
+To achieve this goal effectively, you should consider generating subquestions that will help break down the main question into smaller, more manageable parts. These subquestions might involve understanding the context using the [TOOLS] available, the individuals involved, and the specific details related to goal question.
```

Comment:
A goal is referred to but this goal is not clear yet. Only later, the goal is introduced in line 16-17. Consider moving this text to after the goal.

Also, the second sentence could be adjusted to:
These subquestions might involve understanding the context, the individuals involved and the specific details related to the goal question. You can use the [TOOLS] available to answer the subquestions.